tests/ui/type-alias-impl-trait/alias-bounds-when-bounds-escape.rs

```diff
@@ -0,0 +1,16 @@
+// check-pass
+
+trait Foo {
+    type Assoc<'a, 'b>: 'static;
+}
+
+struct MentionsLifetimeAndType<'a, T>(&'a (), T);
+
+fn foo<'a, 'b, T: Foo>(_: <T as Foo>::Assoc<'a, 'b>) {}
+
+fn test<'b, T: Foo>() {
+    let y: MentionsLifetimeAndType<'_, for<'a> fn(<T as Foo>::Assoc<'a, 'b>)> =
+        MentionsLifetimeAndType(&(), foo);
+}
+
+fn main() {}
```

Comment:
this test doesn't contain an opaque. Please add a comment referencing #117455 and one or two sentences explaining what this is testing